src/lib.rs

```diff
@@ -18,7 +17,15 @@ impl Task for AsyncReflink {
     type JsValue = JsNumber;
 
     fn compute(&mut self) -> Result<Self::Output> {
-        match reflink_copy::reflink(&self.src, &self.dst) {
+        let src_str = self.src.to_str().ok_or_else(|| {
+          Error::from_reason("Invalid UTF-8 sequence in source path".to_string())
+        })?;
+
+        let dst_str = self.dst.to_str().ok_or_else(|| {
+          Error::from_reason("Invalid UTF-8 sequence in destination path".to_string())
+        })?;
```

Comment:
This part of the code would not exist if `copy_on_write` accepts `Path` as parameters.